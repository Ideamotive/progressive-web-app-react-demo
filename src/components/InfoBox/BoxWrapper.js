import React from 'react';
import styled from 'styled-components';

export default styled.div`
    position: relative;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    margin-top: 10px;
    display: flex;
    top: 150px;
    left: 0;
    width: 300px;
    border-radius: 5px;
    flex-direction: column;
    background-color: #fff; 
    box-shadow: 0px 1px 4px 0px rgba(0,0,0,0.75);
    z-index: 10;
    `;
