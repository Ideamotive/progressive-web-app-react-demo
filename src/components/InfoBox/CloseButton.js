import React from 'react';
import styled from 'styled-components';

export default styled.div`
    position: absolute;
    z-index: 10;
    top: 10px;
    right: 10px;
    color: #fff;
    cursor: pointer;
`