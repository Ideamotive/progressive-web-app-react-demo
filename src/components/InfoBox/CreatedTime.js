import React from 'react';
import styled from 'styled-components';

export default styled.div`
    font-weight: bold;
    font-size: 14px;
    padding: 5px 10px;
    text-align: left;
`