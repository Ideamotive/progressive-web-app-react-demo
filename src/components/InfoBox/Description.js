import React from 'react';
import styled from 'styled-components';


export default styled.div`
    margin: 0 0;
    font-size: 18px;
    line-height: 1.5em;
    padding: 5px 10px 10px;
    text-align: justify;
    border-top: 0;
    `;
