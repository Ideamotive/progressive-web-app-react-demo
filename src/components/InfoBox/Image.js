import React from 'react';
import styled from 'styled-components';

export default styled.div`
    width: 100%;
    height: 300px;
    background-size: cover;
    background-image: ${props => props.backgroundImage};
    `;