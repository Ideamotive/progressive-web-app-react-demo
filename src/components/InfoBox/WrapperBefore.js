import React from 'react';
import styled from 'styled-components';

export default styled.div`
    &:before {
        display: inline-block;
        width: 0;
        height: 0;
        border-style: solid;
        border-width: 10px 10px 0 10px;
        border-color: #6980fe transparent transparent transparent;
        line-height: 0;
        _border-color: #6980fe #000000 #000000 #000000;
        _filter: progid:DXImageTransform.Microsoft.Chroma(color='#000000');
    }
`;