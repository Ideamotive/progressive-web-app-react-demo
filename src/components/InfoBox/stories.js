import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text} from '@storybook/addon-knobs/react';

import { withThemeDecorator } from '../storybook/withThemeDecorators';

import InfoBox from './InfoBox';

withThemeDecorator(storiesOf)('InfoBox', module)
    .addDecorator(withKnobs)
    .add('Post Description', () => {
        return (
            <InfoBox
                created_at={new Date()}
                photoDesc={text('Image Description', 'Fake description')}
                imgUrl={text('Image url', 'https://placeimg.com/200/200/any')}
            />)
    });
