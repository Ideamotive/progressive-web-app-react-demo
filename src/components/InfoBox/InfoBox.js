import React, {Component} from 'react';
import PropTypes from 'prop-types';
import moment from 'moment'

// Icon
import 'font-awesome/css/font-awesome.css';

// StyledComponents
import BoxWrapper from './BoxWrapper';
import WrapperBefore from "./WrapperBefore";
import Image from "./Image";
import Description from "./Description";
import CloseButton from "./CloseButton";
import CreatedTime from "./CreatedTime";

class InfoBox extends Component {
    render(){
      if(!this.props) return null;
      const {onBoxClose, mainText, imgUrl, created_at} = this.props;
        return(
            <BoxWrapper>
                <WrapperBefore>
                    <CloseButton onClick={onBoxClose} className='fa fa-times-circle fa-2x' />
                    <Image backgroundImage={`url(${imgUrl})`}/>
                    <CreatedTime>{moment(created_at).format('lll')}</CreatedTime>
                    <Description>{mainText}</Description>
                </WrapperBefore>
            </BoxWrapper>
        )
    }
}

InfoBox.propTypes = {
    onBoxClose: PropTypes.func.isRequired,
    mainText: PropTypes.string.isRequired,
    imgUrl: PropTypes.string.isRequired,
    created_at: PropTypes.oneOfType([PropTypes.string, PropTypes.instanceOf(Date)])
};

InfoBox.defaultProps = {
    mainText: '',
    created_at: new Date('01-01-20018'),
    onBoxClose: () => {},
    imgUrl: '',
};


export default InfoBox;
