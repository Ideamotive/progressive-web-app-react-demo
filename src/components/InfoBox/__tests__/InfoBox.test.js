import React from 'react';
import * as renderer from 'react-test-renderer';

import InfoBox from '../InfoBox';

describe('<InfoBox/> component', () => {
  test('should match to snapshot', () => {
    const tree = renderer
      .create(
        <InfoBox/>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

