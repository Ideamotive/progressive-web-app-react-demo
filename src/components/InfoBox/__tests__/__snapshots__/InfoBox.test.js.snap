// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoBox/> component should match to snapshot 1`] = `
<div
  className="sc-bdVaJa dxyeMJ"
>
  <div
    className="sc-bwzfXH gtIphb"
  >
    <div
      className="fa fa-times-circle fa-2x sc-ifAKCX icZIwo"
      onClick={[Function]}
    />
    <div
      className="sc-htpNat fKOxhJ"
    />
    <div
      className="sc-EHOje gVvQOi"
    >
      Jan 1, 20018 12:00 AM
    </div>
    <div
      className="sc-bxivhb cTZUEo"
    >
      
    </div>
  </div>
</div>
`;
